lib/matplotlib/_api/__init__.py

```diff
@@ -361,3 +361,43 @@ def warn_external(message, category=None):
             break
         frame = frame.f_back
     warnings.warn(message, category, stacklevel)
+
+
+def remove_methods(methods, cls=None):
+    """
+    Class decorator for removing methods on subclasses.
+
+    Use as ::
+
+        @_api.remove_methods(['methA', 'methB'])
+        class C: ...
+
+    Because almost everything in Matplotlib inherent from
+    `matplotlib.artist.Artist` we have some inherited methods
+    that do not function correctly on the sub-classes.  To prevent
+    this from catching users who expect something to work from the
+    docs this will remove them from the docs and cause the methods
+    to error when used.
+    """
+    if cls is None:  # Return the actual class decorator.
+        return functools.partial(remove_methods, methods)
+
+    def make_wrapper(name):
```

Comment:
```suggestion
    def make_wrapper(method_name):
```

Reply:
oh, it must have been doing terrible closure + scope things before....